### PR TITLE
docs: release process: RPC documentation

### PR DESCRIPTION
**Pull request by karelbilek**
The auto-generated RPC docs seem to work so far ( https://bitcoincore.org/en/doc/ + 0.17.0 https://github.com/bitcoin-core/bitcoincore.org/pull/618 ). There are some problems (the design of the list on the right is not ideal, and apparently the huge list of pages slows down jekyll), but that can be fixed later; people are already linking to the docs now and looking for them there

So I am adding the RPC docs to the release process.

The script is here and it is written in golang, since I am most confident in the language; if necessary, I can try to rewrite to python, which is more common in bitcoin tooling

https://github.com/bitcoin-core/bitcoincore.org/tree/master/contrib/doc-gen

### CODE REVIEW

File: doc/release-process.md [hunk 1]
```diff
@@ -295,6 +295,8 @@ bitcoin.org (see below for bitcoin.org update instructions).
   - Bitcoin Core announcements list https://bitcoincore.org/en/list/announcements/join/
 
   - bitcoincore.org blog post
+
```
Comment (MarcoFalke) [inline doc/release-process.md]: nit: trailing whitespace

**Review by maflcko — APPROVED** [review]

**Review by fanquake — APPROVED** [review]

**Comment by promag** [comment]
ACK.

**Review by ken2812221 — APPROVED** [review]
ACK 3b706212ad319423147c9ccc681556c052050186

**Comment by laanwj** [comment]
Thanks for making this,

though, I don't think this change is useful without a further description of *how* to do this release process step, not even a reference to this issue.
(unless this is a note to yourself)

**Comment by karel-3d** [comment]
Good point.

But I am not sure if info about that should be added rather here or rather into bitcoin-core/bitcoincore.org git repo. (It is currently hidden in the contrib folder.)

**Comment by laanwj** [comment]
sure, but then there should be a link to that at least, I suppose?

**Pull request merged**